lms/djangoapps/letstudy_api/views.py

```diff
@@ -0,0 +1,165 @@
+"""
+API end-point for creating a new user without password.
+"""
+import logging
+from uuid import uuid4
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from opaque_keys import InvalidKeyError
+from social_django.models import UserSocialAuth
+
+from enrollment.api import add_enrollment, get_enrollment
+from enrollment.views import ApiKeyPermissionMixIn
+from lms.djangoapps.instructor.views.tools import get_student_from_identifier
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from student.models import UserProfile
+from third_party_auth.models import OAuth2ProviderConfig
+
+log = logging.getLogger(__name__)
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    Create user account.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    _error_dict = {
+        "username": "Username is required parameter.",
+        "email": "Email is required parameter.",
+        "uid": "Uid is required parameter."
+    }
+
+    def _check_available_required_params(self, parameter, parameter_name, values_list=None):
+        """
+        Check required parameter is correct.
+        If parameter isn't correct ValueError is raised.
+        :param parameter: object
+        :param parameter_name: string. Parameter's name
+        :param values_list: List of values
+        :return: parameter
+        """
+        if not parameter or (values_list and isinstance(values_list, list) and parameter not in values_list):
+            raise ValueError(self._error_dict[parameter_name].format(value=parameter))
+        return parameter
+
+    def post(self, request):
+        """
+        Create user by the email and the username.
+        """
+        data = dict(request.data.iteritems())
```

Comment:
Where is all this data dict used? I see it's collected, but don't see it's applied

Reply:
fixed